go/stats/prombackend/collectors.go

```diff
@@ -0,0 +1,214 @@
+package prombackend
+
+import (
+	"strings"
+
+	prom "github.com/prometheus/client_golang/prometheus"
+	"vitess.io/vitess/go/stats"
+)
+
+// metricsCollector collects both stats.Counters and stats.Gauges
+type metricsCollector struct {
+	counters map[*stats.Counters]*prom.Desc
```

Comment:
From what I can tell reading below, it looks like we never actually put more than one `stats.Counters` into any given metricsCollector, so can this be simplified to just have a pointer to the counter and a pointer to the `Desc`?